File: The-Golden-Leaf-server/src/main/java/com/example/datban/repository/BanSlotRepository.java
package com.example.datban.repository;

import com.example.datban.model.BanSlot;
import org.springframework.data.jpa.repository.JpaRepository;

import java.time.LocalDate;
import java.util.Optional;

public interface BanSlotRepository extends JpaRepository<BanSlot, Long> {
    Optional<BanSlot> findByNgayAndKhungGio(LocalDate ngay, String khungGio);
}

File: Datban/app/src/main/java/com/example/giaodien/data/repository/BanSlotRepository.kt
package com.example.giaodien.data.repository

import com.example.giaodien.data.model.BanSlot
import com.example.giaodien.data.network.RetrofitInstance

class BanSlotRepository {

    suspend fun getBanSlots(): List<BanSlot> {
        return RetrofitInstance.api.getBanSlots()
    }

    suspend fun reserveBanSlot(ngay: String, khungGio: String, soLuong: Int): BanSlot {
        return RetrofitInstance.api.reserveBanSlot(ngay, khungGio, soLuong)
    }
}
